bitnami/postgresql-ha/templates/postgresql/hooks-scripts-configmap.yaml

```diff
@@ -63,11 +63,21 @@ data:
     primary_host="${primary_node[0]}"
 
     # Stop postgresql for graceful exit.
+    PG_STOP_TIME=$EPOCHSECONDS
     postgresql_stop
 
     if [[ -z "$primary_host" ]] || [[ "$primary_host" == "$REPMGR_NODE_NETWORK_NAME" ]]; then
         info "Primary node need to wait for a new primary node before leaving the cluster"
         retry_while is_new_primary_ready 10 5
     else
-        info "Standby node doesn't need to wait, leaving the cluster."
+        info "Standby node doesn't need to wait for a new primary switchover"
+    fi
+
+    # Make sure pre-stop hook waits at least 25 seconds after stop of PG to make sure PGPOOL detects node is down.
+    # default terminationGracePeriodSeconds=30 seconds
+    PG_STOP_DURATION=$(($EPOCHSECONDS - $PG_STOP_TIME))
+    if (( $PG_STOP_DURATION < 25 )); then
```

Comment:
I think the `25` should be parametrized, other users may need a different delay.
Could you add a setting for this ?

Reply:
Setting added